website/blog/2023-11-09-sharing-podman-images-with-kubernetes-cluster.md

```diff
@@ -0,0 +1,197 @@
+
+Oh wait, it means that no more registry, additional steps would be required ? just build it and load it.
+
+## Minikube to the rescue
+
+While we have the goal of using both cri-o and podman altogether, we can explore the current projects allowing us to quickly setup kubernetes clusters.
+
+### kind
+
+On the `kind` side, there is a default configuration that is using containerd and there is no plan to support an alternative such as cri-o https://github.com/kubernetes-sigs/kind/issues/1369#issuecomment-867440704
+
+That said, some people try to maintain a way to do that but not officialy https://gist.github.com/aojea/bd1fb766302779b77b8f68fa0a81c0f2
+
+By doing that, we would also need to mount `/var/lib/containers` folder from the host (the podman machine) to the container. And there is no easy flag in kind.
+
+### Minikube
+
+#### Minikube options
+
+Minikube supports more options and provides a way to select the container engine runtime. And it includes support for cri-o.
+
+We can use the `container-runtime` parameter and ask for `cri-o` runtime. Command-line should include `--container-runtime=cri-o`.
+
+Then, we do have a podman/cri-o environment inside the container and can use `eval $(minikube podman-env)` to use on the Windows/macOS podman CLI.
+
+One issue is that we have then two 'podman engine', one which is running inside the podman machine and another one running inside the container. And the podman included in the container is using version 3.4 as it's on Debian/Ubuntu stable while recent version of podman is 4.7.x.
+
+Can we mount the podman machine `/var/lib/containers` into the container ?
+
+yes ! minikube provides some options to do additional mount with `--mount-string` argument. It is not obvious but you also need to add the `--mount` in addition to this parameter. Full argument is `--mount --mount-string "<host-path:container-path>"`
+
+But `/var` is already a mounted folder. So here the idea is to change the path of where cri-o is storing its data.
+
+So we can provide a custom mounted path and cri-o use that custom location. Let's pickup `/host-containers`.
+
+When starting minikube we need then to add `--mount --mount-string "/var/lib/containers:/host-containers"`.
+
+How to configure cri-o. For now it's not possible using minikube options. But minikube allows you to change the base image.
+
+#### Minikube kicbase image
+
+Let's do our own base image named kicbase image.
+
+Minikube includes a default configuration file for cri-o.
+https://github.com/kubernetes/minikube/blob/0b29983f4bdc1ad55180ee43e3f34cae6c24dee4/deploy/kicbase/02-crio.conf
+
+Let's include the configuration part in this file.
+
+```toml
+[crio]
+root = "/host-containers/storage"
+runroot = "/host-containers/storage"
+```
+
+Let's upgrade as well podman that is inside the container adding the instruction
+
+instead of
+
+```Dockerfile
+RUN clean-install podman && \
```

Comment:
I think there is a missing reference to the original Dockerfile that we should modify in order to generate our own image